### PR TITLE
Add support for Kafka topics

### PR DESCRIPTION
**Pull request by dweinshenker**
**What**

- Add support for kafka topics based on godo updates (https://github.com/digitalocean/godo/pull/633)
- Fix UptimeAlert marshalling for struct vs. string (https://github.com/digitalocean/godo/pull/637)

**Testing**

Ran `TestAccDigitalOceanUptimeAlert_Basic` and `TestAccDigitalOceanDatabaseKafkaTopic` acceptance tests locally

### CODE REVIEW

File: digitalocean/database/resource_database_kafka_topic.go [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+)
+
+func ResourceDigitalOceanDatabaseKafkaTopic() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanDatabaseKafkaTopicCreate,
+		ReadContext:   resourceDigitalOceanDatabaseKafkaTopicRead,
+		UpdateContext: resourceDigitalOceanDatabaseKafkaTopicUpdate,
+		DeleteContext: resourceDigitalOceanDatabaseKafkaTopicDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDigitalOceanDatabaseUserImport,
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_kafka_topic.go:23]: Looks like this needs an import function.  The database user import is a good example for a resource like this where it needs more than one identifier to import (ie the parent cluster's ID and the name).

https://github.com/digitalocean/terraform-provider-digitalocean/blob/fb899442bee65259dcc3810f5cde963dd4c79009/digitalocean/database/resource_database_user.go#L190

**Review by andrewsomething — CHANGES_REQUESTED** [review]

File: digitalocean/database/resource_database_kafka_topic.go [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("Error retrieving kafka topic: %s", err)
+	}
+
+	d.Set("state", topic.State)
+	d.Set("replication_factor", topic.ReplicationFactor)
+	d.Set("partitions", topic.Partitions)
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_kafka_topic.go:367]: Similar issue here:

```
panic: Invalid address to set: []string{"partitions"}

goroutine 729 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc000995e80, {0x113e468, 0xa}, {0xf736c0, 0xc0008ed938})
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:230 +0x2b5
github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database.resourceDigitalOceanDatabaseKafkaTopicRead({0x146b5b8, 0xc0008eb260}, 0x37?, {0xff81e0?, 0xc0004bb320?})
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database/resource_database_kafka_topic.go:371 +0x25e
```

File: digitalocean/database/resource_database_kafka_topic.go [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("Error retrieving kafka topic: %s", err)
+	}
+
+	d.Set("state", topic.State)
```
[the remark this answers is not in the document]
Comment (dweinshenker) [reply digitalocean/database/resource_database_kafka_topic.go:369]: Fixed

File: digitalocean/database/resource_database_kafka_topic_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+const testAccCheckDigitalOceanDatabaseKafkaTopicComplete = `
+%s
+
+resource "digitalocean_database_kafka_topic" "foobaz" {
+  cluster_id         = digitalocean_database_cluster.foobar.id
+  name               = "%s"
+  partition_count    = %d
+  replication_factor = %d
+  config = {
```
[the remark this answers is not in the document]
Comment (dweinshenker) [reply digitalocean/database/resource_database_kafka_topic_test.go]: Fixed

File: digitalocean/database/resource_database_kafka_topic.go (hunk 1 above)
Comment (dweinshenker) [reply digitalocean/database/resource_database_kafka_topic.go]: Fixed

File: digitalocean/database/resource_database_kafka_topic.go (hunk 2 above)
Comment (dweinshenker) [reply digitalocean/database/resource_database_kafka_topic.go]: Fixed, I removed `partitions` as it is not relevant for configuration but rather shows state on each partition. It is probably not relevant for the TF client

**Review by andrewsomething — APPROVED** [review]
:+1:  LGTM!

**Pull request merged**